src/components/Terminal/ProblemsTab.tsx

```diff
@@ -1,73 +1,331 @@
+    });
+  }, [restrictedDiagnostics, search]);
 
   return (
     <div className="flex-1 text-white text-xs h-full relative overflow-x-hidden overflow-y-auto">
-      {errors.length ? (
-        <div className="px-4">
-          <table className="border-collapse w-full">
-            <thead>
-              <tr>
-                <th className="p-2 w-8">Line</th>
-                <th className="p-2 text-left">Title</th>
-                <th className="p-2 text-left">Details</th>
-              </tr>
-            </thead>
-            <tbody>
-              {errors.map((err: any, id) => {
-                const { title, detail, location } = err;
-                let renderedLine = err.location?.startLine;
-                renderedLine = renderedLine && err.location?.startColumn ? `${renderedLine}:${err.location?.startColumn}` : renderedLine;
-                return (
-                  <tr key={title || id} className="border-t border-gray-700">
-                    <td
-                      className="p-2 cursor-pointer"
-                      onClick={() =>
-                        NavigationService.scrollToEditorLine(
-                          location?.startLine || 0,
-                          location?.startColumn,
-                        )
-                      }
-                    >
-                      {renderedLine || '-'}
-                    </td>
-                    <td className="p-2 text-left">{title}</td>
-                    <td className="p-2 text-left">{detail || '-'}</td>
-                  </tr>
-                );
-              })}
-            </tbody>
-          </table>
+      <div className="px-4 pt-2">
+        <div className='pb-2 flex flex-row items-center'>
+          <SeverityButtons diagnostics={restrictedDiagnostics} active={active} setActive={setActiveFn} />
+          <div className='ml-2 flex-1 flex flex-row items-center justify-center rounded-md border border-transparent shadow-xs px-2 py-1 bg-gray-700 text-xs font-medium'>
+            <VscSearch />
+            <input ref={inputRef} placeholder='Filter diagnostics...' className='w-full bg-gray-700 border-transparent ml-2 focus:border-transparent focus:ring-0 focus:outline-none' onChange={debounce((e) => setSearch(e.target.value), 250)} />
+            <button type='button' className={`hover:bg-gray-900 rounded-sm border border-transparent ${search ? 'opacity-100' : 'opacity-0'}`} onClick={() => {
+              if (inputRef.current) {
+                inputRef.current.value = '';
+              }
+              setSearch('');
+            }}>
+              <VscClose />
+            </button>
+          </div>
+          <a href='https://stoplight.io/open-source/spectral' title='Spectral website' target='_blank' rel="noreferrer" className='text-white hover:text-blue-500 mx-2'>
+            <span>
+              Powered by
+            </span>
+            {' '}
+            <strong>
+              Spectral
+            </strong>
+          </a>
```

Comment:
I'm curious. Why do we have this here? 🤔 It looks like an ad 😅

Reply:
Removed.